src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -46,6 +46,11 @@ internal abstract class AnalyzerDriver : IDisposable
         private Task _primaryTask;
 
         /// <summary>
+        /// Number of worker tasks processing compilation events and executing analyzer actions.
+        /// </summary>
+        private readonly int _workerCount = Math.Min(Environment.ProcessorCount, 10);
```

Comment:
@JohnHamby For simplicity, I have kept this as ProcessorCount, bounded to 10. This change didn't regress the build time on machine.